app/src/main/kotlin/com/simplemobiletools/smsmessenger/helpers/ImportExportProgressNotification.kt

```diff
@@ -0,0 +1,116 @@
+package com.simplemobiletools.smsmessenger.helpers
+
+import android.app.*
+import android.content.Context.NOTIFICATION_SERVICE
+import androidx.core.app.NotificationCompat
+import com.simplemobiletools.commons.helpers.isOreoPlus
+import com.simplemobiletools.smsmessenger.R
+import com.simplemobiletools.smsmessenger.activities.SimpleActivity
+import java.util.*
+
+class ImportExportProgressNotification(
+    private val activity: SimpleActivity,
+    private val type: ImportOrExport
+){
+    enum class ImportOrExport{
+        IMPORT, EXPORT
+    }
+    private var isInit = false
+    private lateinit var notification: NotificationCompat.Builder
+    private lateinit var channelId: String
+    private val progressMax = 100
+    private val notificationManager = activity.getSystemService(NOTIFICATION_SERVICE) as NotificationManager
+    private var waitTime = 0L
+    private var startTime = 0L
+
+    init {
+        startTime = System.currentTimeMillis()
+        if (isOreoPlus()) {
+            channelId = activity.getString(R.string.channel_import_export)
+            val name = activity.getString(R.string.channel_import_export)
+            val importance = NotificationManager.IMPORTANCE_LOW
+            val mChannel = NotificationChannel(channelId, name, importance)
+
+            notificationManager.createNotificationChannel(mChannel)
+        }
+    }
+
+    fun setFinish(hasSucceed: Boolean) {
+        if (isInit)
+        {
+            isInit = false
+            notification.setContentTitle(if(type == ImportOrExport.EXPORT) activity.getString(R.string.exporting_successful) else activity.getString(R.string.importing_successful))
+            if(hasSucceed)
+            {
+                notification.setContentText("Success")
+                    .setProgress(0,0, false)
+                    .setOngoing(false)
+                    .setAutoCancel(true)
+            }
+            else
+            {
+                notification.setContentText("Something went wrong")
+                    .setProgress(0,0, false)
+                    .setOngoing(false)
+                    .setAutoCancel(true)
+            }
+            notificationManager.notify(EXPORT_IMPORT_NOTIFICATION_ID, notification.build())
+        }
+    }
+
+    fun spawnProgressNotification()
+    {
+        val contentTitle = if(type == ImportOrExport.IMPORT) activity.getString(R.string.importing_messages) else activity.getString(R.string.exporting_messages)
```

Comment:
shorten such lines please by moving some part of it into some variable and always add spaces after "if"

Reply:
Done, I've rejected else block for the string to next line, idk if that following coding convention or not...